cirq/contrib/noise_models/noise_models.py

```diff
@@ -142,3 +145,121 @@ def noisy_moment(self, moment: 'cirq.Moment',
                 moment,
                 ops.Moment(self.qubit_noise_gate(q) for q in system_qubits)
             ]
+
+
+class PerQubitDepolarizingWithDampedReadoutNoiseModel(devices.NoiseModel):
```

Comment:
One comment on the direction of this approach. I think it would be better to make different noise models composable, instead of making a different noise model for each combination of error A, B and C. In other words, it would probably be better to use DepolarisingNoiseModel on your circuit, then DampedReadoutNoiseModel on the resulting moments.

Reply:
My concern with this is (AFAICT) composing noise models can result in duplicated noise channels. This is illustrated in the example below:
```
create circuit C with gates {X, H, M}
create a noise model K which adds k_err before M gates, apply K to C
-> C is now {X, H, k_err, M}
create a noise model D which adds d_err before non-M gates, apply D to C
-> C is now {X, d_err, H, d_err, k_err, d_err, M}
```
where the final `d_err` is not desired, but will be added since we have no clear way to differentiate between "gates representing noise" and "gates representing real operations." (We might have this behavior and I've simply missed it - please point it out if this is the case.)

Reply:
It looks like this is relevant to issue #1835 - @Strilanc @mpharrigan, do we expect any issues with the method above? If not, let's apply that fix and I'll come back to this PR afterward.

Reply:
Yes, that's the relevant issue. If someone wants to tackle it; that would be great. One of the reasons why these noise models are in `contrib` instead of cirq proper is because we'd ideally like to solve the problem of composable noise models and just use that

Reply:
From group discussions it sounds like this is a good use case for TaggedOperations (see #1940, #2670). Opened #2719 to track this change - I'll revisit this PR once that issue is resolved.

Reply:
On further inspection, I've determined that noise models will need significant work before supporting composition in a consistent and intuitive way. In order to avoid delaying the noise-from-metrics feature any further, I'm moving forward with this PR.

After the noise-model work is done, we can split this monolithic model up into its component pieces and compose them as necessary.